src/test/java/spoon/MavenLauncherTest.java

```diff
@@ -44,9 +48,23 @@
 
 public class MavenLauncherTest {
 
+	@AfterEach
+	void tearDown() throws IOException {
+		// Clean up old classpath files to ensure tests actually re-run properly
+		try (Stream<Path> paths = Files.walk(Path.of("src/test/resources/maven-launcher"))) {
+			List<Path> filesToDelete = paths
+				.filter(it -> it.getFileName().toString().equals("spoon.classpath.tmp"))
+				.collect(Collectors.toList());
+
+			for (Path path : filesToDelete) {
+				Files.deleteIfExists(path);
+			}
+		}
+	}
```

Comment:
I think it would be better if the tests were executed in a temporary directory instead (e.g. using `@TempDir` from JUnit5).

Reply:
This is something of a systematic error we're doing (i.e. just running in the local directory), but perhaps this PR would be a good driving force to actually start running tests appropriately isolated.

Reply:
Yea, that was a bit lazy… In exchange I have converted all launcher tests (that aren't disabled) to use temporary folders. The diff is a bit more annoying now but I hope still readable enough.